### PR TITLE
Intro overhaul

### PR DESCRIPTION
**Pull request by Crell**
This supplants #473, which was a PR from a fork. This brings the branch back into the repo for easier collaboration.

### CODE REVIEW

File: src/overview/how-it-works.md [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+Every time you push to a live branch (a git branch with an active environment attached to it) or activate an [environment](administration/web/environments.md) for a branch, there are two main processes that happen: *Build* and *Deploy*.  
+
+1.  The build process looks through the configuration files in your repository and assembles the necessary containers.  
+2. The deploy process makes those containers live, replacing the previous versions, with virtually no interruption in service.
 
 ![The Build and Deploy pipeline](/images/build-pipeline.svg)
 
+## Good Practices and Constraints
+
+Platform.sh promotes the practice of using a build-oriented structure; This  means that instead of committing to your Git repository the final state of your application, you would only commit the files specific to it, any frameworks, libraries and dependencies should be pulled at build-time as should any assets that should be compiled.
```
Comment (Crell) [inline src/overview/how-it-works.md]: Given that in, eg, Go committing dependencies is still very common and in fact necessary (until the new generation of dependency management tools is fully adopted), we should not push that line as hard.  The best practices here vary language to language.

**Comment by Crell** [comment]
I did some cleanup and revised some of the language from Ori's PR.  On the whole I'm OK with these changes, and I love the revised diagrams.

The biggest change was the Basics page, which felt too markety.  The documentation should be teaching, not selling.  I brought back some of the language and concepts from the current version of that page and tried to blend the two together, but it doesn't fully flow in my mind yet.  Suggestions there welcome, but bear in mind the target audience here is developers who have already decided Platform.sh is worth using or at least digging into.

File: src/overview/how-it-works.md (hunk 1 above)
Comment (OriPekelman) [inline src/overview/how-it-works.md]: This is still the good practice. Commit a lock file, not the libs. Add a comment if you'd like that sometimes you don't have a choice. Sometimes you vendorize. That's ok too.

File: src/overview/how-it-works.md [hunk 2]
```diff
@@ -15,15 +15,14 @@ Platform.sh promotes the practice of using a build-oriented structure; This  mea
...
 
->Most projects have a single `.platform.app.yaml` file  Platform.sh supports multiple applications in a single project, it will scan the repository for `.platform.app.yaml` in subdirectories and will build each subdirectory that contains one as an independent application (the built application will not contain any directories above the one in which it is found); The system is smart enough not to rebuild applications that have already been built, so if you have multiple applications, only the one that changed will be rebuilt and redeployed.
+> Most projects have a single `.platform.app.yaml` file  Platform.sh supports multiple applications in a single project.  It will scan the repository for `.platform.app.yaml` files in subdirectories and will build each subdirectory that contains one as an independent application. The built application will not contain any directories above the one in which it is found. The system is smart enough not to rebuild applications that have already been built, so if you have multiple applications, only the one that changed will be rebuilt and redeployed.
```
Comment (OriPekelman) [inline src/overview/how-it-works.md]: "While most projects.."

**Comment by OriPekelman** [comment]
Anyway, content fine by me.  I am on a bad connection;  So I will let someone else verify nothing is broken functionally. We still want to add some diagrams, quickly.

**Comment by JGrubb** [comment]
I didn't do a full regression test, but I love the changes that have been made here.  +1 from me.

**Pull request merged**